modules/platforms/dotnet/Apache.Ignite.Core/Impl/Client/ClientSocket.cs

```diff
@@ -182,6 +203,43 @@ internal sealed class ClientSocket : IDisposable
             TaskRunner.Run(WaitForMessages, TaskCreationOptions.LongRunning);
         }
 
+        /// <summary>
+        /// Gets the heartbeat interval according to server-side and client-side configuration.
+        /// </summary>
+        private TimeSpan GetHeartbeatInterval(IgniteClientConfiguration clientConfiguration)
+        {
+            var serverIdleTimeoutMs = DoOutInOp(
+                ClientOp.GetIdleTimeout, null, r => r.Reader.ReadLong());
+
+            // ReSharper disable once PossibleLossOfFraction
+            var recommendedHeartbeatInterval = TimeSpan.FromMilliseconds(serverIdleTimeoutMs / 3);
+
+            if (recommendedHeartbeatInterval > TimeSpan.Zero)
+            {
+                if (clientConfiguration.HeartbeatInterval < recommendedHeartbeatInterval)
+                {
+                    _logger.Info(
+                        $"Server-side IdleTimeout is {serverIdleTimeoutMs}ms, " +
+                        $"using configured {nameof(IgniteClientConfiguration)}." +
+                        $"{nameof(IgniteClientConfiguration.HeartbeatInterval)}: " +
+                        clientConfiguration.HeartbeatInterval);
+
+                    return clientConfiguration.HeartbeatInterval;
+                }
+
+                _logger.Info($"Server-side IdleTimeout is {serverIdleTimeoutMs}ms, " +
+                             $"using 1/3 of it as heartbeat interval: {recommendedHeartbeatInterval}");
```

Comment:
This should be a `Warn` in my opinion as we definitely have a problem with client/server configuration here and potentially unexpected behavior for the user.

Reply:
Agree, fixed.